smart-send-logistics/includes/frontend/class-ss-shipping-endpoint.php

```diff
@@ -0,0 +1,126 @@
+<?php
+
+if (!defined('ABSPATH')) {
+    exit; // Exit if accessed directly
+}
+
+/**
+ * Class SS_Shipping_Api_Endpoint
+ *
+ * Handles the REST API endpoints for Smart Send Logistics.
+ *
+ * This class is responsible for registering the REST API endpoints and handling
+ * requests to fetch the closest pickup points based on provided address parameters.
+ *
+ * @package    SmartSendLogistics
+ * @subpackage SS_Shipping_Endpoint
+ * @category   REST API
+ * @author     Smart Send
+ */
+class SS_Shipping_Api_Endpoint
+{
+    /**
+     * @var FrontendInterface $frontend An instance of the frontend class handling agent fetching and session management.
+     */
+    protected $frontend;
+
+    /**
+     * SS_Shipping_Endpoint constructor.
+     *
+     * @param SS_Shipping_Frontend $frontends An instance of a class implementing the frontend logic.
+     */
+    public function __construct(SS_Shipping_Frontend $frontends)
+    {
+        $this->frontend = $frontends;
+        add_action('rest_api_init', array($this, 'register_endpoints'));
+    }
+
+    /**
+     * Registers the custom REST API endpoint for fetching closest pickup points.
+     *
+     * @return void
+     */
+    public function register_endpoints()
+    {
+        // Register the API endpoint
+        register_rest_route('smart-send-logistics/v1', '/get-closest-pickup-points', array(
+            'methods' => 'POST',
+            'callback' => array($this, 'get_pickup_points_callback'),
+            'permission_callback' => "__return_true",
+        ));
+        // Register the API endpoint for shipping carrier
+        register_rest_route('smart-send-logistics/v1', '/get-shipping-carrier', array(
+            'methods' => 'POST',
+            'callback' => array($this, 'get_shipping_carrier'),
+            'permission_callback' => "__return_true",
+        ));
+    }
+
+    /**
+     * Callback function for the REST API endpoint to fetch pickup points.
+     *
+     * @param WP_REST_Request $request The request object containing parameters.
+     * @return WP_REST_Response Response object with pickup points or an error message.
+     */
+    public function get_pickup_points_callback(WP_REST_Request $request)
+    {
+        $country = wc_clean($request->get_param('country'));
+        $postal_code = wc_clean($request->get_param('postCode'));
+        $city = (!empty($request->get_param('city')) ? wc_clean($request->get_param('city')) : null);
+        $street = wc_clean($request->get_param('street'));
+        $carrier = wc_clean($request->get_param('carrier'));
+        $is_rest_api = true;
+
+        $ss_agents = $this->frontend->handle_agents_session($carrier, $country, $postal_code, $city, $street, $is_rest_api);
```

Comment:
Why do we need to do *anything* with the session when using the Rest API?

Reply:
We are not using the sessions for the front end, we are using them to get the formatted address of agent when agent id is passed from front end to backend.

Reply:
Is that the best way to do it? Why not return all the required data from the frontend to the backend?

What do we do in the current code? We don't use session to store the pickup point addresses do we?

Reply:
In the classic WooCommerce checkout process, we utilize sessions to store and retrieve details about the selected pickup point. This approach is necessary because the order details require more information than just the address and agent number—additional contextual data related to the pickup point is also needed. For the same reason, we employ session management within WooCommerce blocks to ensure that this critical information is consistently accessible and accurately reflected throughout the checkout experience.